src/hparams.jl

```diff
@@ -0,0 +1,126 @@
+import .tensorboard_plugin_hparams.hparams: Interval, MetricInfo, MetricName, HParamInfo, Experiment, HParams
+import .tensorboard_plugin_hparams.hparams: var"#DataType" as HParamDataType, DatasetType as HDatasetType
+import .tensorboard_plugin_hparams.google.protobuf: ListValue as HListValue, OneOf as HOneOf, Value as HValue
+import .tensorboard_plugin_hparams.hparams as HP
+import .tensorboard: DataClass
+
+struct HParamRealDomain
+    min_value::Float64
+    max_value::Float64
+end
+struct HParamSetDomain{T<:Union{String, Bool, Float64}}
+    values::Vector{T}
+end
+Base.@kwdef struct HParamConfig
+    name::String
+    datatype::DataType
+    displayname::String = ""
+    description::String = ""
+    domain::Union{Nothing, HParamRealDomain, HParamSetDomain} = nothing
+end
+Base.@kwdef struct MetricConfig
+    name::String
+    displayname::String = ""
+    description::String = ""
+end
+
+default_domain(::Val{Bool}) = HParamSetDomain([false, true])
+default_domain(::Val{Float64}) = HParamRealDomain(typemin(Float64), typemax(Float64))
```

Comment:
Shouldn't this be more flexible?
```suggestion
default_domain(f::Val{AbstractFloat}) = HParamRealDomain(typemin(f), typemax(f))
```

Reply:
It seems that the hparams plugin only supports `String`, `Bool` or `Float64`. I have took this suggestion and made the API allow any `Real` number which is then cast to a `Float64` and explained this in the docstring.